test/controllers/AdminControllerSpec.scala

```diff
@@ -259,7 +259,30 @@ class AdminControllerSpec
       }
     }
 
-    "must must state to rejected when state is Submitted and Approved request is sent" in {
+    "must return not found when state is not Submitted or Rejected, and update request is sent" in {
+      val notSubmittedState = Approved("file-name", Year.parse("2024"))
+      val submission = aSubmission.copy(state = notSubmittedState)
+      val app: Application = GuiceApplicationBuilder().overrides(
+        bind[SubmissionRepository].toInstance(mockSubmissionRepo),
+        bind[BackendAuthComponents].toInstance(backendAuthComponentsStub)
+      ).build()
+
+      when(mockStubBehaviour.stubAuth(any, eqTo(Retrieval.EmptyRetrieval))).thenReturn(Future.unit)
+      when(mockSubmissionRepo.getById("some-submission-id")).thenReturn(Future.successful(Some(submission)))
+      when(mockSubmissionRepo.save(any)).thenReturn(Future.successful(Done))
+
+      running(app) {
+        val request = FakeRequest(routes.AdminController.updateBlockedSubmission("some-submission-id"))
+          .withHeaders("Authorization" -> "Token some-token")
+          .withJsonBody(Json.toJson(UpdateSubmissionStateRequest("unknown")))
+        val result = route(app, request).value
+
+        status(result) mustBe NOT_FOUND
+      }
+      verify(mockSubmissionRepo, never).save(any)
+    }
+
+    "must state to approved when state is Submitted and Approved request is sent" in {
```

Comment:
"must **set** state to **Approved** when state is Submitted and Approved request is sent"